src/robotFactory.js

```diff
@@ -1,10 +1,69 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+    this.coords = {};
+    this.coords.x = position.x ? position.x : 0;
+    this.coords.y = position.y ? position.y : 0;
```

Comment:
You can simplify it even more using logical operator `||`:
```suggestion
    this.coords = {
      x: position.x || 0,
      y: position.y || 0,
    };
```